packages/office-ui-fabric-react/src/components/DetailsList/examples/DetailsList.CustomFooter.Example.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { TextField } from 'office-ui-fabric-react/lib/TextField';
+import {
+  DetailsList,
+  DetailsListLayoutMode,
+  Selection,
+  IColumn,
+  IDetailsList,
+  IDetailsFooterProps
+} from 'office-ui-fabric-react/lib/DetailsList';
+import { MarqueeSelection } from 'office-ui-fabric-react/lib/MarqueeSelection';
+import { IRenderFunction, createRef } from 'office-ui-fabric-react/lib/Utilities';
+import { Checkbox } from 'office-ui-fabric-react/lib/Checkbox';
+const _items: any[] = [];
+
+const _columns: IColumn[] = [
+  {
+    key: 'column1',
+    name: 'Name',
+    fieldName: 'name',
+    minWidth: 100,
+    maxWidth: 200,
+    isResizable: true,
+    ariaLabel: 'Operations for name'
+  },
+  {
+    key: 'column2',
+    name: 'Value',
+    fieldName: 'value',
+    minWidth: 100,
+    maxWidth: 200,
+    isResizable: true,
+    ariaLabel: 'Operations for value'
+  }
+];
+
+export class DetailsListCustomFooterExample extends React.Component<
+  {},
+  {
+    items: {}[];
+    selectionDetails: {};
+    showItemIndexInView: boolean;
+  }
+> {
+  private _selection: Selection;
+  private _detailsList = createRef<IDetailsList>();
+
+  constructor(props: {}) {
+    super(props);
+
+    // Populate with items for demos.
+    if (_items.length === 0) {
+      for (let i = 0; i < 200; i++) {
+        _items.push({
+          key: i,
+          name: 'Item ' + i,
+          value: i
+        });
+      }
+    }
+
+    this._selection = new Selection({
+      onSelectionChanged: () => this.setState({ selectionDetails: this._getSelectionDetails() })
+    });
+
+    this.state = {
+      items: _items,
+      selectionDetails: this._getSelectionDetails(),
+      showItemIndexInView: false
+    };
+  }
+
+  public render(): JSX.Element {
+    const { items, selectionDetails } = this.state;
+
+    return (
+      <div>
+        <div>{selectionDetails}</div>
+        <div>
+          <Checkbox
+            label="Show index of the first item in view when unmounting"
+            checked={this.state.showItemIndexInView}
+            onChange={this._onShowItemIndexInViewChanged}
+          />
+        </div>
+        <TextField label="Filter by name:" onChanged={this._onChanged} />
+        <MarqueeSelection selection={this._selection}>
+          <DetailsList
+            componentRef={this._detailsList}
+            items={items}
+            columns={_columns}
+            setKey="set"
+            layoutMode={DetailsListLayoutMode.fixedColumns}
+            selection={this._selection}
+            selectionPreservedOnEmptyClick={true}
+            ariaLabelForSelectionColumn="Toggle selection"
+            ariaLabelForSelectAllCheckbox="Toggle selection for all items"
+            onItemInvoked={this._onItemInvoked}
+            onRenderDetailsFooter={
```

Comment:
Hoping we can replace this with something like: 

onRenderDetailsFooter={
  (detailsFooterProps: IDetailsFooterProps, defaultRender: IRenderFunction<IDetailsFooterProps>) => (                
              <div>
                  {defaultRender({
                    footerText: 'Details List Footer',
                    ...detailsFooterProps
                  })}
                </div>
              )
}